pkg/gitreceive/build.go

```diff
@@ -81,11 +82,14 @@ func build(conf *Config, builderKey, gitSha string) error {
 	buildDir := filepath.Join(repoDir, "build")
 
 	slugName := fmt.Sprintf("%s:git-%s", appName, shortSha)
-	imageName := strings.Replace(slugName, ":", "-", -1)
 	if err := os.MkdirAll(buildDir, os.ModeDir); err != nil {
 		return fmt.Errorf("making the build directory %s (%s)", buildDir, err)
 	}
-	tmpDir := os.TempDir()
+
+	tmpDir, err := ioutil.TempDir(buildDir, "tmp")
```

Comment:
I believe that this change will mean that _every_ `git push` (regardless of history) has to push everything to builder. We should try to keep it in one directory, or at least the same place per project